lesson28/src/js/register.js

```diff
@@ -110,6 +110,17 @@ const saveUserData = () => {
         password : password.value,
     }
 
-    window.localStorage.setItem("userData", JSON.stringify(userData));
-    window.location.href = "register-done.html";
+    if( isRegisteredEmail(email) ){
+        validation.showErrorMessage(email,"登録済みのメールアドレスです");
+    }else{
+        window.localStorage.setItem("userData", JSON.stringify(userData));
+        window.location.href = "register-done.html";
+    }
+}
+
+const isRegisteredEmail = (email) => {
+    const userData = JSON.parse(window.localStorage.getItem("userData"));
+    const registeredEmail = userData?.email;
+
+    return registeredEmail === email.value;
 }
```

Comment:
Returns whether the value entered matches the value of localStrage.
Checking for the existence of userData in local storage and checking if userData has an email property is done with the `userData?.email` optional chain.

Reply:
It is a very simple way to learn how to write!